transformers/events/log_value/transformer.go

```diff
@@ -17,67 +17,48 @@
 package log_value
 
 import (
-	"fmt"
 	"math/big"
 
 	"github.com/makerdao/vdb-mcd-transformers/transformers/shared/constants"
 
-	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/makerdao/vdb-mcd-transformers/transformers/shared"
 	"github.com/makerdao/vulcanizedb/libraries/shared/factories/event"
 	"github.com/makerdao/vulcanizedb/pkg/core"
 	"github.com/makerdao/vulcanizedb/pkg/datastore/postgres"
-	"github.com/makerdao/vulcanizedb/pkg/eth"
 )
 
 type Transformer struct{}
 
 const Val event.ColumnName = "val"
 
-func (Transformer) toEntities(contractAbi string, logs []core.EventLog) ([]LogValueEntity, error) {
-	var entities []LogValueEntity
+func (t Transformer) ToModels(abi string, logs []core.EventLog, db *postgres.DB) ([]event.InsertionModel, error) {
+	var models []event.InsertionModel
 	for _, log := range logs {
-		var entity LogValueEntity
-		address := log.Log.Address
-		abi, parseErr := eth.ParseAbi(contractAbi)
-		if parseErr != nil {
-			return nil, parseErr
-		}
 
-		contract := bind.NewBoundContract(address, abi, nil, nil, nil)
-		unpackErr := contract.UnpackLog(&entity, "LogValue", log.Log)
-		if unpackErr != nil {
-			return nil, unpackErr
+		err := shared.VerifyLog(log.Log, shared.OneTopicRequired, shared.LogDataRequired)
+		if err != nil {
+			return nil, err
 		}
 
-		entity.HeaderID = log.HeaderID
-		entity.LogID = log.ID
-		entities = append(entities, entity)
-	}
-
-	return entities, nil
-}
-
-func (t Transformer) ToModels(abi string, logs []core.EventLog, _ *postgres.DB) ([]event.InsertionModel, error) {
-	entities, entityErr := t.toEntities(abi, logs)
-	if entityErr != nil {
-		return nil, fmt.Errorf("transformer couldn't convert logs to entities: %v", entityErr)
-	}
+		contractAddressID, contractAddressErr := shared.GetOrCreateAddress(log.Log.Address.String(), db)
+		if contractAddressErr != nil {
+			return nil, err
+		}
 
-	var models []event.InsertionModel
-	for _, logValueEntity := range entities {
-		bigIntVal := new(big.Int).SetBytes(logValueEntity.Val[:])
+		bigIntVal := new(big.Int)
+		bigIntVal.SetBytes(log.Log.Data)
```

Comment:
mostly just curious - what inspired you to take out parsing of the entity from the ABI? This seems more elegant - I hadn't even realized we could do this 🎉

Reply:
😄 I'm glad it works, I think I just noticed that during the refactor that if all I needed was the data, I could grab it from the log.